src/main/java/seedu/address/logic/commands/AddHouseCommand.java

```diff
@@ -0,0 +1,117 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_BLOCK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_HOUSING_TYPE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LEVEL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_POSTALCODE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRICE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STREET;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_UNITNUMBER;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.house.House;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Seller;
+
+/**
+ * Adds a seller to the address book.
+ */
+public class AddHouseCommand extends Command {
+
+    public static final String COMMAND_WORD = "addHouse";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a house to EstateEase. Indicate N/A for "
+            + "nonexistent fields. "
+            + "Parameters: "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_HOUSING_TYPE + "HOUSING_TYPE "
+            + PREFIX_STREET + "STREET "
+            + PREFIX_BLOCK + "BLOCK "
+            + PREFIX_LEVEL + "LEVEL "
+            + PREFIX_UNITNUMBER + "UNIT NUMBER "
+            + PREFIX_POSTALCODE + "POSTAL CODE "
+            + PREFIX_PRICE + "PRICE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_HOUSING_TYPE + "Condominium "
+            + PREFIX_STREET + "Clementi Ave 2 "
+            + PREFIX_BLOCK + "N/A "
+            + PREFIX_LEVEL + "02 "
+            + PREFIX_UNITNUMBER + "25 "
+            + PREFIX_POSTALCODE + "578578 "
+            + PREFIX_PRICE + "99999 ";
+
+    public static final String MESSAGE_SUCCESS = "New house added!";
+    public static final String MESSAGE_DUPLICATE_SELLER = "This house already exists in EstateEase";
+
+    public static final String MESSAGE_INVALID_SELLER = "This Seller does not exist in EstateEase";
+
+    private final House houseToAdd;
+
+    private final Name nameToCheck;
+
+    /**
+     * Creates an AddHouseCommand to add the specified {@code house}
+     */
+    public AddHouseCommand(House house, Name name) {
+        requireNonNull(house);
```

Comment:
I think you need to check nonNull for name too? or can name be actually null?

Reply:
Resolved in future PR